system/system_monitor/reader/hdd_reader/hdd_reader.cpp

```diff
@@ -89,15 +87,15 @@ typedef struct
   uint8_t command_;             //!< @brief COMMAND
   uint8_t reserved2_;           //!< @brief Reserved
   uint8_t control_;             //!< @brief CONTROL
-} ATAPassThrough12;
+};
 
 /**
  * @brief Attribute Table Format
  * @note For details please see the documents below.
  * - SMART Attribute Overview
  *   http://www.t13.org/Documents/UploadedDocuments/docs2005/e05171r0-ACS-SMARTAttributes_Overview.pdf
  */
-typedef struct __attribute__((packed))  // Minimize total struct memory 16 to 12
+using AttributeEntry = struct __attribute__((packed))  // Minimize total struct memory 16 to 12
```

Comment:
How about using `#pragma pack`? Is it different from `__attribute__((packed))`?

Reply:
The usage of `#pragma pack(1)` is better style, I think.

Reply:
The `__attribute__((packed))` might be a GCC-specific in ancient times.
Nowadays, we can use it in Clang as well as GCC. Seems to be no problem.
https://wandbox.org/permlink/p1fWzK5JqEVVDnaJ
Also, AUTOSAR C++14 says `#pragma` should not be used (Rule A16-7-1).

Reply:
Understood. Looks like it should stay as it is! :+1: